core/cache/model.go

```diff
@@ -560,14 +560,9 @@ func (m *Model) removeBranch(ch RemoveBranch) error {
 func (m *Model) setDetails(details ModelChange) {
 	m.mu.Lock()
 
-	// If this is the first receipt of details, set the removal message.
-	if m.removalMessage == nil {
-		m.removalMessage = RemoveModel{
-			ModelUUID: details.ModelUUID,
-		}
-	}
-
-	m.setStale(false)
+	m.setRemovalMessage(RemoveModel{
```

Comment:
There's a lock acquisition on the line above that deadlocks this call.

Reply:
There are 2 muexes. One on the Model and one on the embedded Resident struct.
The one used in the setRemoveMessage is the Resident one.